sdk/js/src/utils/parseVaa.ts

```diff
@@ -27,49 +48,35 @@ export function parseNFTPayload(payload: Buffer) {
   };
 }
 
+/**
+ * TokenTransferPayload is the payload data for a Token transfer VAA.
+ */
+export type TokenTransferPayload = Pick<TokenTransfer, "amount" | "fee"> & {
+  originAddress: string;
+  originChain: number;
+  targetAddress: string;
+  targetChain: number;
+  fromAddress?: string;
+};
+
 //     0   u256     amount
 //     32  [u8; 32] token_address
 //     64  u16      token_chain
 //     66  [u8; 32] recipient
 //     98  u16      recipient_chain
 //     100 u256     fee
-export function parseTransferPayload(payload: Buffer) {
+export function parseTransferPayload(payload: Buffer): TokenTransferPayload {
   const parsed = parseTokenTransferPayload(payload);
   return {
     amount: parsed.amount,
     originAddress: parsed.tokenAddress.toString("hex"),
     originChain: parsed.tokenChain,
     targetAddress: parsed.to.toString("hex"),
     targetChain: parsed.toChain,
-    fee: parsed.fee === null ? undefined : parsed.fee,
+    fee: parsed.fee === null ? null : parsed.fee,
```

Comment:
Is there a reason for changing this from `undefined` to `null`? Would prefer to keep it as `undefined` if possible to not break anyone not expecting `null`.

Reply:
The reason was because the `TokenTransfer` uses `fee: bigint | null` and since `TokenTransferPayload` uses fee it would make sense to use the same type for consistency. But yea this might be breaking so can revert to undefined.